### DISCUSSION TITLE
Moving from custom code-server install to module causes startup_script issue

### DISCUSSION THREAD

**Discussion by TheZoker** [coder/coder#12063] [General] [answered]
Hi there,

Yesterday I switched form installing the code-server directly in the Dockerfile to using the module:
https://registry.coder.com/modules/code-server

```yml
module "code-server" {
  source   = "registry.coder.com/modules/code-server/coder"
  version  = "1.0.2"
  agent_id = coder_agent.main.id
  display_name = "VS Code"
  install_version = "4.21.0"
  folder = "/home/${local.username}/workspace"
  port = "8080"
  extensions = [
    "eamodio.gitlens",
    "PKief.material-icon-theme"
  ]
}
```

I then saw, that my `coder_agent` 
```yml
resource "coder_agent" "main" {
  arch           = data.coder_provisioner.me.arch
  os             = "linux"
  startup_script = "code-server"
  dir            = "/home/${local.username}/workspace/"
}
```
produces this error when I start the container:
```
/bin/bash: line 1: code-server: command not found
```

So I remove the `startup_script` part from the `coder_agent`, but now I cannot connect to it via the `VS Code Desktop` button anymore. It loads the windows, but loads for ever:

![image](https://github.com/coder/coder/assets/1368405/c521205b-44fa-4abe-a5e9-eeff681fe873)

And when I open the terminal app, it shows this:

![image](https://github.com/coder/coder/assets/1368405/115cbe89-47aa-489d-9deb-02bf7a562cee)

How can I fix this? Do I need the `startup_script` anyway? Why does it cause this error? Thanks!

**Comment by matifali** [comment, marked as answer]
You should remove `startup_script = "code-server"`, module will also start the code-server process

**Reply by TheZoker** [reply]
Yes that's what I already wrote in the initial issue. But then I'm not able to connect to it via the VSCode Desktop anymore. 

Anyway I already solved the issue, seemed to be an issue with my `personalize` file. When I removed it, it started again

**Reply by matifali** [reply]
VS Code desktop does not have anything related to `startup_script`, so the issue you were seeing was probably the personalise script.

**Reply by TheZoker** [reply]
The strange thing is, when I did not use the module, both worked fine. But since it's working now, I will mark this issue as solved
